modules/redis/redis.c

```diff
@@ -246,57 +252,40 @@ redis_worker_insert(LogThreadedDestDriver *s, LogMessage *msg)
       return LTR_NOT_CONNECTED;
     }
 
-  log_template_format(self->key, msg, &self->template_options, LTZ_SEND,
-                      self->super.worker.instance.seq_num, NULL, self->key_str);
-
-  if (self->param1)
-    log_template_format(self->param1, msg, &self->template_options, LTZ_SEND,
-                        self->super.worker.instance.seq_num, NULL, self->param1_str);
-  if (self->param2)
-    log_template_format(self->param2, msg, &self->template_options, LTZ_SEND,
-                        self->super.worker.instance.seq_num, NULL, self->param2_str);
+  argv = g_malloc(argc * sizeof(char *));
+  argvlen = g_malloc(argc * sizeof(size_t));
 
   argv[0] = self->command->str;
   argvlen[0] = self->command->len;
-  argv[1] = self->key_str->str;
-  argvlen[1] = self->key_str->len;
-
-  if (self->param1)
-    {
-      argv[2] = self->param1_str->str;
-      argvlen[2] = self->param1_str->len;
-      argc++;
-    }
 
-  if (self->param2)
-    {
-      argv[3] = self->param2_str->str;
-      argvlen[3] = self->param2_str->len;
-      argc++;
-    }
+  gint index = 1;
+  gpointer user_data[] = { self, msg, argv, argvlen, &index };
+  g_list_foreach(self->fields, _fill_argv, user_data);
 
   reply = redisCommandArgv(self->c, argc, argv, argvlen);
 
+  GString *full_command = g_string_new(argv[0]);
```

Comment:
This is a per-message allocation but the string is only used for debug/error messages.
I'd use a scratch-buffer or a truncated buffer on the stack.

Reply:
Thanks, will do!

Reply:
Done, thanks!